test/e2e/transaction_queue.ts

```diff
@@ -141,45 +140,39 @@ describe('Transaction Queue', () => {
         });
 
         it('should enable job processor', async () => {
-            await agenda.enable({ name: 'processWithdrawals' });
+            await agenda.enable({ name: eventNameProcessWithdrawals });
         });
 
         it('should cast a fail event', (done) => {
             const callback = async (error: Error, job: Job) => {
-                agenda.off('fail:processWithdrawals', callback);
-
-                const { withdrawal } = await WithdrawalService.getById(job.attrs.data.documentId);
-
-                expect(withdrawal).toBeDefined();
-                expect(withdrawal.withdrawalId).toBeUndefined();
-                expect(withdrawal.failReason).toBeUndefined();
-
+                agenda.off(`fail:${eventNameProcessWithdrawals}`, callback);
+                expect(error.message).toBe(ERROR_MAX_FEE_PER_GAS);
                 done();
             };
-            agenda.on('fail:processWithdrawals', callback);
+            agenda.on(`fail:${eventNameProcessWithdrawals}`, callback);
+            agenda.now(eventNameProcessWithdrawals, null);
```

Comment:
@bram-rongen  I was under the impression that this would create a new job and run it immediately, resulting in 2 jobs with the same processor being repeated on a different timeframe. If that is not the case this might also be a good one to implement in the controllers that schedule a withdrawal. Since:

* If the job is not running, agenda should start processing the single scheduled withdrawal instead of waiting for the 5s cycle to start. This might take max 5s in the worst case. But would be improved to instant if its "in between job runs":)

* If the job is running, agenda should just do nothing and let the withdrawal be picked up in the next run, since it can only run one job at a time.

Agreed?

Reply:
Not sure, but might break the event listeners in the tests, maybe those are a bit time dependent currently. Moving agenda.enable() to after agenda.on() should fix that.

Reply:
yes, love it!